src/libraries/Microsoft.Extensions.Primitives/tests/StringSegmentTest.cs

```diff
@@ -1280,6 +1280,12 @@ public void TrimEnd_RemovesTrailingWhitespaces(string value, int start, int leng
         [MemberData(nameof(GlobalizationCompareTestData))]
         public void StringSegment_CompareEqual_Globalized(string a, string b, StringComparison comparisonType, int expectedCompareToSign)
         {
+            if ((comparisonType == StringComparison.InvariantCulture || comparisonType == StringComparison.InvariantCultureIgnoreCase) 
+                && (PlatformDetection.IsiOS || PlatformDetection.IstvOS))
+            {
+                return;
```

Comment:
Can you add a comment as to why these types are being skipped on these OSes?  Same for the cases below.

Reply:
I don't think this is the correct fix because it doesn't just apply to iOS/tvOS but all cases where we run in Invariant Globalization Mode.

You can change the `expectedComparetToSign` in those cases by modifying the test data and checking for `PlatformDetection.IsInvariantGlobalization`, similar to how we do it in https://github.com/dotnet/runtime/blob/d77092decb2bbd7bc94cd548af925d801a032e9e/src/libraries/Common/tests/Tests/System/StringTests.cs#L493-L494

Reply:
Thank you, Alexander!